drivers/hyperv/hyperv.go

```diff
@@ -172,21 +177,32 @@ func (d *Driver) Create() error {
 		return err
 	}
 
-	log.Infof("Creating VM...")
-	virtualSwitch, err := d.chooseVirtualSwitch()
-	if err != nil {
-		return err
-	}
-
-	log.Infof("Using switch %q", virtualSwitch)
-
-	if err := cmd("Hyper-V\\New-VM",
+	args := []string{
+		"Hyper-V\\New-VM",
 		d.MachineName,
 		"-Path", fmt.Sprintf("'%s'", d.ResolveStorePath(".")),
-		"-SwitchName", quote(virtualSwitch),
-		"-MemoryStartupBytes", toMb(d.Memory)); err != nil {
+		"-MemoryStartupBytes", toMb(d.Memory),
+	}
+	if d.VirtualSwitch != "" {
+		log.Infof("Creating VM...")
+		virtualSwitch, err := d.chooseVirtualSwitch()
+		if err != nil {
+			return err
+		}
+		log.Infof("Using switch %q", virtualSwitch)
+		args = append(args, "-SwitchName", quote(virtualSwitch))
+	}
+
+	if err := cmd(args...); err != nil {
 		return err
 	}
+
+	if d.VirtualSwitch == "" {
+		if err := cmd("Hyper-V\\Remove-VMNetworkAdapter", "-VMName", d.MachineName); err != nil {
```

Comment:
@guillaumerose this logic is here just to remove the crc named network adapter in case o there is no virtualswitch provided? Is this virtualswitch present by default?

Reply:
`Hyper-V\New-VM` always attach a network adapter. I didn't found an argument to disable it. I had to remove it by hand just after.

Reply:
Attach a network adapter as name of the machine? @gbraad you have any idea around it?

Reply:
New-vm might enforce an interface, just not aware this was the case. Will check

Reply:
there is no way around this